File: Datasets/TrickyBugs/GenProgs/dpp_generated_progs_python/p04000/p04000_num5_parsed.py
def compute_subrectangles(H, W, N, cells):
    counts = [0] * 10
    
    # create a 2D grid with H rows and W columns
    grid = [[0] * W for _ in range(H)]
    
    # paint the cells with black
    for i in range(N):
        r, c = cells[i]
        grid[r-1][c-1] = 1
    
    # count the number of subrectangles with different number of black cells
    for r in range(H-2):
        for c in range(W-2):
            black_count = 0
            for i in range(3):
                for j in range(3):
                    if grid[r+i][c+j] == 1:
                        black_count += 1
            counts[black_count] += 1
    
    return counts


# get the input
H, W, N = map(int, input().split())
cells = [tuple(map(int, input().split())) for _ in range(N)]

# compute the subrectangles
counts = compute_subrectangles(H, W, N, cells)

# print the counts
for count in counts:
    print(count)

File: Datasets/TrickyBugs/GenProgs/dpp_generated_progs_cpp/p04000/p04000_num5_parsed.cpp
#include <iostream>
#include <vector>

using namespace std;

int main() {
    int H, W, N;
    cin >> H >> W >> N;

    vector<vector<int>> grid(H, vector<int>(W, 0));

    for (int i = 0; i < N; i++) {
        int row, col;
        cin >> row >> col;
        grid[row - 1][col - 1] = 1;
    }

    vector<int> count(10, 0);

    for (int r = 0; r < H - 2; r++) {
        for (int c = 0; c < W - 2; c++) {
            int blackCount = 0;
            for (int i = r; i < r + 3; i++) {
                for (int j = c; j < c + 3; j++) {
                    if (grid[i][j] == 1) {
                        blackCount++;
                    }
                }
            }
            count[blackCount]++;
        }
    }

    for (int i = 0; i < 10; i++) {
        cout << count[i] << endl;
    }

    return 0;
}